core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -185,6 +211,10 @@ private[spark] class PipedRDD[T: ClassTag](
       }
 
       private def cleanup(): Unit = {
+        // interrupt the stdin writer thread, so the stdin writer thread for ShuffledRDD could be
```

Comment:
Not necessary. This will happen first if command is failed/exited, then interrupt in task completion listener is redundant.

This is my initial fix. Will remove it.